pkg/s3wrapper/filesystem.go

```diff
@@ -55,22 +55,62 @@ func (f *FSClient) CreateBucket() error {
 
 const xattrUserAttributePrefix = "user."
 
+func isUnsupportedXattrError(err error) bool {
+	xattr_error, ok := err.(*xattr.Error)
+	if !ok {
+		return true
+	}
+	errno, ok := xattr_error.Err.(syscall.Errno)
+	if !ok {
+		return true
+	}
+	if errno == syscall.ENOTSUP || errno == syscall.EOPNOTSUPP {
+		return true
+	}
+	return false
+}
+
 func (f *FSClient) writeFileMetadata(filePath string, metadata map[string]string) error {
+	if xattr.XATTR_SUPPORTED == false {
+		f.log.Warn("Xattr is not supported for your platform - consider upgrading to one that supports xattr")
+		return nil
+	}
 	for attributeName, attributeValue := range metadata {
 		err := xattr.Set(filePath, strings.ToLower(fmt.Sprintf("%s%s", xattrUserAttributePrefix, attributeName)), []byte(attributeValue))
-		if err != nil {
+		if err != nil && !isUnsupportedXattrError(err) {
 			return errors.Wrapf(err, "unable to store metadata key %s = %s", attributeName, attributeValue)
 		}
+		// Warn that Xattr is unsupported in the filesystem for the path - then ignore the error
+		f.log.Warnf(
+			"Error while setting xattr %s=%s for path %s, the filesystem does not support extended attributes (xattr) - consider upgrading to one that supports xattr",
+			fmt.Sprintf("%s%s", xattrUserAttributePrefix, attributeName),
+			attributeValue,
+			filePath,
+		)
```

Comment:
This will generate a log message each time we try to write or read a file, right? Won't that be too much logging? Can we instead do a check when the FSClient object is created, write the log message, save the result in flag, and then avoid the xattr operation (and the log message) when it is true?

Reply:
I think the main issue with this is that the way we detect Xattr enablement ism with the use of an actual file/path, we respond to the exception when we attempt to write.

On creation of the FSClient, we don't have the context required to check this.

Reply:
I went for adding a flag to limit the warning messages to a single message.